### ISSUE TITLE
Foliate can't open epubs from the Library

### ISSUE THREAD

**Issue by marcioccamargo** [johnfactotum/foliate#1209]
**Describe the bug**
I can't open EPUBs that are added to my catalog.

**To Reproduce**
1.  Open Nautilus.
2.  Find an .epub file and open it with Foliate.
3.  You can read normally, but when you return to the library, all books are inaccessible. When I try to open one epub, a message appears: "The book cannot be opened."

**Expected behavior**

**Screenshots**

**Version:**
 - Foliate version: 3.1.0
 - OS/Distribution and version: FEDORA 39
 - Desktop environment: Gnome 45.2
 - Installation method: Flatpak

**Additional context**
I tried this with all my epubs, they are stored locally. Foliate opens the file normally when I double click  the file in nautilus, but when I try to continue to read a epub in the library, it doesn't work.

Here is a video:

[Gravação de tela de 2024-01-05 18-58-46.webm](https://github.com/johnfactotum/foliate/assets/24537073/87d55d12-d2d0-4751-a345-a6db9d72c092)

**Comment by marcioccamargo** [comment]
A better video:

[Gravação de tela de 2024-01-05 19-03-21.webm](https://github.com/johnfactotum/foliate/assets/24537073/6830bf3b-ac27-4843-a99c-66bc17eee08f)

**Comment by RuggeroRebellato** [comment]
Same here.

**Comment by ConEstArthur** [comment]
Facing the same issue.
I mentioned the same in #1178
but I was only facing it in files that I open from file explorer and not "Open File" option from Foliate.
However recently I have noticed I face the same issue with files that had been opened from Foliate itself.

I am on the same platform as you are on.

**Comment by crimsonfall** [comment]
That's odd.

I have faced this same issue in previous versions of Foliate, while also on Fedora (KDE) no less. I thought that the app's rewrite would have fixed it.

**Comment by marcioccamargo** [comment]
> That's odd.
> 
> I have faced this same issue in previous versions of Foliate, while also on Fedora (KDE) no less. I thought that the app's rewrite would have fixed it.

Same for me.

**Comment by johnfactotum** [comment]
This particular bug is a regression in v3.1.0. For some reason I can't seem to reproduce it when running it with GNOME Builder, which is why the bug made it to the release.

Anyway, to fix this properly, it needs to get the permissions somehow, either by granting filesystem permissions, or by somehow using the documents portal directly. Though it would be nice if Tracker has an API for this.

For now a workaround would be either adding filesystem permissions for the locations indexed by Tracker, or removing permission for Tracker all together.

Probably the easiest fix would be to add `--filesystem=xdg-documents:ro` and `--filesystem=xdg-download:ro`. That would fix #942/#1178 (provided that your files are in those locations and indexed by Tracker).

But it might be better to have a bug fix release that simply removes the Tracker policy. Because Tracker never worked for the Flatpak version of Foliate, anyway.

**Comment by ConEstArthur** [comment]
> Facing the same issue. I mentioned the same in #1178 but I was only facing it in files that I open from file explorer and not "Open File" option from Foliate. However recently I have noticed I face the same issue with files that had been opened from Foliate itself.
> 
> I am on the same platform as you are on.

facing the same thing again ; foliate does not remember location of the books I have in my library after a restart of the system.
I get "Book cannot be opened" for every book in my collection. #1178 

Edit:- It works now I do not know how..

**Comment by marcioccamargo** [comment]
I just upgraded to Foliate 3.1.1(from flathub), but the problem persists. I am sending a video showing the issue. For some reason, one of the epubs is opening without a problem. I don't know what is going on. There is nothing different about this particular epub; they are all in the same directory with the exception of "Eloquent Javascript."

After I deleted all epubs from the library, they started to open normally again. However, after I closed Foliate, the problem started again.

[Gravação de tela de 2024-04-04 12-19-20.webm](https://github.com/johnfactotum/foliate/assets/24537073/0b691492-70f5-4733-95f0-64d1917d9530)

**Comment by johnfactotum** [comment]
Yes, that's #942 (which is the same as #1178), which is different from this issue (#1209). 

To clarify, #942 is the issue where the file is no longer exported by the portal after a while. You can work around #942 using Tracker, which will report the path of the file to Foliate. However it can only get the original path of the file, which Foliate cannot access from the sandbox, causing *this* issue, #1209, where you can't open files from the library even if they are still available from the portal.

In other words, #1209 was caused by an incomplete solution to #942. #1209 was fixed by unfixing #942.

To actually fix #942, one could add the following permissions:
- `--add-policy=Tracker3.dbus:org.freedesktop.Tracker3.Miner.Files=tracker:Documents`
- `--filesystem=xdg-documents:ro`, `--filesystem=xdg-download:ro`, etc. depending on where the files are located

**Comment by marcioccamargo** [comment]
I got it, thanks!

**Comment by FabianEscarate** [comment]
hi. i don´t know if it's the same problem but i did this step (previuly i install folliate from flathub).
1.- install Flatseal (form flathub)
2.- add this Variable in "Variable seccion" 
      WEBKIT_DISABLE_DMABUF_RENDERER=1
3.- try again to open foliate
